README.md

```diff
@@ -1,12 +1,93 @@
-An email segmentation system:
-- reference implementation of ECIR 2018 paper
+QuaggaLib was developed to provide an easy-to-use library for the model we describe in our ECIR paper (see below). 
+For code, data, and competitor approaches used in the paper, please check out [our other repository](https://github.com/HPI-Information-Systems/Quagga).
 
-# Reference
+![](docs/email_segmentation.png)
+
+QuaggaLib parses the raw email body into separate blocks and extracts meta-data from inline-headers.
+This kind of pre-processing should be used in all applications using email data.
+The library provides the actual written text content as well as the meta-data that would otherwise be hidden in the unstructured email body.
+
+![](docs/line_classification.png)
+
+The Quagga model classifies each line into one of five types: Header, Body, Body/Intro, Body/Outro, Body/Signature.
+We also provide models which only distinguish Header and Body.
+
+![](docs/dl_model.png)
+
+At the heart of Quagga, we use a neural network.
+For more information, we refer to our paper presented at ECIR 2018.
+QuaggaLib wraps this model and adds modular processing steps to read *.eml files, a web frontend and server, a parser that uses the line classifications to form blocks, and a parser that extracts structured data from Header blocks.
+
+## Usage
+### Installation
+```bash
+git clone git@github.com:HPI-Information-Systems/QuaggaLib.git
+cd QuaggaLib
+python setup.py install
+```
+
+Note, that QuaggaLib currently only works with Python 3.6!
+
+### QuaggaServer
+QuaggaServer provides a web interface for easy manual testing.
+To use the interface, run 
+
+```bash
+python QuaggaServer/QuaggaServer.py
+```
+
+... and open https://localhost:5000/index in your favourite browser.
+
+The interface is self-explanatory.
+We provide multiple pre-trained models in QuaggaLib.
+You can select a model that was trained on the Enron corpus or our ASF corpus.
+Furthermore you can switch between 2-zone or 5-zone classification.
+By toggling the CRF/CNN switch, you can use either the line encoder result or that after the full model.
+
+### QuaggaLib for line classification
+```python
+from Quagga import Quagga, 
+                   EmailDirectoryReader, 
+                   ListReaderRawEmailTexts, 
+                   ListReaderExtractedBodies, 
+                   TempQuaggaReader,
+                   ModelBuilder
+from Quagga.Utils.BlockParser.BlockParser import BlockParser
+
+quagga = Quagga(ListReaderExtractedBodies(''), 
+                output_dir='', # optional, saves to directory if set
+                model_builder=ModelBuilder(with_crf=True, zones=2, trainset='enron'), # optional
+                block_parser=BlockParser()) # optional
+
+classified_lines = quagga.predict(email_body_text)
+```
+
+You can use the `ListReaderExtractedBodies([body1, body2,..])` to iterate over a bulk of email bodies.
```

Comment:
not in this scope but as i'm reading this i think that we maybe should choose better names for these classes. any ideas?